File: euser-gateway/src/main/java/com/example/eusergateway/exchange/EuserSSOExchange.java
package com.example.eusergateway.exchange;

import com.example.eusergateway.dto.response.R;
import com.example.eusergateway.exchange.response.PersonalPermissionInfo;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.service.annotation.GetExchange;
import org.springframework.web.service.annotation.HttpExchange;

@HttpExchange
public interface EuserSSOExchange {

    @GetExchange("/user/permission/all")
    R<PersonalPermissionInfo> getPersonalAllPermissions(
            @RequestHeader("X-Euser") String username
    );
}

File: api-factory/af-gateway/src/main/kotlin/com/example/afgateway/exchange/EuserSSOExchange.kt
package com.example.afgateway.exchange

import com.example.afgateway.exchange.response.QueryPublicAPIPermissionResponse
import com.example.afgateway.exchange.response.R
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.service.annotation.GetExchange
import org.springframework.web.service.annotation.HttpExchange
import reactor.core.publisher.Mono

@HttpExchange(url = "/euser-sso/internal")
interface EuserSSOExchange {

    @GetExchange("/perms/public-api")
    fun queryPublicAPIPerms(@RequestParam("un") username: String): Mono<R<QueryPublicAPIPermissionResponse>>

}